src/lib/graft/connection.cpp

```diff
@@ -82,7 +82,27 @@ void UpstreamSender::send(TaskManager &manager, const std::string& def_uri)
         }
         std::stringstream ss;
         ss << "http://0.0.0.0:" << mg_port << "/callback/" << boost::uuids::to_string(callback_uuid);
-        output.headers.emplace_back(std::make_pair("X-Callback", ss.str()));
+
+        auto it = std::find_if(output.headers.begin(), output.headers.end(), [](auto& v)->bool { v.first == "X-Callback"; } );
+        if(it != output.headers.end())
+        {
+            std::ostringstream oss;
+            oss << "Warning, X-Callback header exists and will be overwritten. '" << it->second << "' will be replaced by '" << ss.str() << "'";
```

Comment:
Probably "warning" should be logger severity, not part of the message?
`MWARNING()` or `LOG_PRINT_L0()` macros

Reply:
if this "header" replacement is not "too bad/exeptional" case - just remove "Warning" word keeping existing log level